File: front-end/src/app/domain/models/game/action/impl/gamePlay/WinGameAction.ts
import {GameAction} from "../../GameAction";
import {Player} from "../../../Player";

/**
 * TODO: Add JsDoc
 *
 * @author Hamza el Haouti
 */
export class WinGameAction extends GameAction {

  public constructor(
    id: number,
    actor: Player,
    performedAt: Date,
  ) {
    super(WinGameAction.name, id, actor, performedAt);
  }

  public static trueCopy(action: WinGameAction): WinGameAction | null {
    return new WinGameAction(
      action.id,
      Player.trueCopy(action.actor),
      new Date(action.performedAt),
    );
  }

  public override toJson(): object {
    return super.toJson();
  }

  public override toString(): string {
    return `The game is finished. ${this.actor.user.username} won.`;
  }

}

File: back-end/src/main/java/nl/hva/backend/domain/models/game/action/impl/gamePlay/WinGameAction.java
package nl.hva.backend.domain.models.game.action.impl.gamePlay;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import nl.hva.backend.domain.models.game.Player;
import nl.hva.backend.domain.models.game.action.GameAction;

import javax.persistence.Entity;

/**
 * TODO: Add JavaDoc
 *
 * @author Hamza el Haouti
 */
@Entity
@Getter
@Setter
@NoArgsConstructor
public class WinGameAction extends GameAction {

    public WinGameAction(
            Player actor
    ) {
        super(actor);
    }

}